src/nisarqa/utils/utils.py

```diff
@@ -480,6 +481,60 @@ def set_logger_handler(
         log.addHandler(handler)
 
 
+def log_function_start_and_stop_time(func):
+    """
+    Function decorator which logs the start and completion of a function.
+
+    The function's arguments are also logged. Useful for benchmarking; the
+    log file can be parsed for timings.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+
+        # Construct the "arguments" string for the log message
+        def trunc(arg) -> str:
+            max_length = 40
+            if isinstance(arg, nisarqa.YamlParamGroup):
+                max_length = 100
+            a = f"{arg!r}"
+            if len(a) < max_length or isinstance(
+                arg, (h5py.Dataset, nisarqa.ComplexFloat16Decoder)
+            ):
+                return arg
+            else:
+                return f"{a[:max_length]}[TRUNCATED]"
+
+        name = ""
+        obj_with_name_attr = (
+            h5py.Dataset,
+            nisarqa.ComplexFloat16Decoder,
+            nisarqa.Raster,
+        )
+        for a in args:
+            if isinstance(a, obj_with_name_attr):
+                name = f"{a.name}. "
+                break
+        if not name:
+            for val in kwargs.values():
+                if isinstance(val, obj_with_name_attr):
+                    name = f"{val.name}. "
+                    break
+        trunc_args = tuple((trunc(i) for i in args))
+        trunc_kwargs = {key: trunc(val) for key, val in kwargs.items()}
+
+        suffix = f"`{func.__name__}`. {name}args={trunc_args}. kwargs={trunc_kwargs}."
```

Comment:
I think the `name` thing here is a bit too kludgy.

If you're logging the runtime of a function that just has one or two salient parameters, I would use the context manager approach instead. This approach where we log all of the args & kwargs is most useful for lower-level functions with fewer parameters.

Let's remove `name` from the string here and instead have something like this:

```suggestion
        suffix = f"Called `{func.__name__}` with args={trunc_args} and kwargs={trunc_kwargs}."
```